traits/tests/test_regression.py

```diff
@@ -309,6 +310,18 @@ class A(HasTraits):
         with self.assertRaises(ZeroDivisionError):
             a.bar = "foo"
 
+    def test_clone_list_of_enum_trait(self):
+        # Regression test for enthought/traits#1622.
+
+        class Company(HasTraits):
+            departments = List(Str)
+            projects = List(Enum(values="departments"))
+
+        company = Company(departments=["sales"], projects=["sales"])
+        clone = company.clone_traits()
+
+        self.assertEqual(clone.projects, ["sales"])
```

Comment:
I would like to suggest a better example that provides a more concrete illustration of the need for this. It is a really minor thing, but I think the naming can clarify things better.
```suggestion
        class Order(HasTraits):
            menu = List(Str)
            selection = List(Enum(values="menu"))

        order = Order(menu=["fish"], selection=["fish"])
        clone = order.clone_traits()

        self.assertEqual(clone.selection, ["fish"])
```